packages/react-components/react-utilities/src/compose/resolveShorthand.ts

```diff
@@ -26,11 +27,17 @@ export const resolveShorthand: ResolveShorthandFunction = (value, options) => {
 
   let resolvedShorthand = {} as UnknownSlotProps;
 
-  if (typeof value === 'string' || typeof value === 'number' || Array.isArray(value) || isValidElement(value)) {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  if (typeof value === 'string' || typeof value === 'number' || Array.isArray(value) || isValidElement<any>(value)) {
     resolvedShorthand.children = value;
   } else if (typeof value === 'object') {
     resolvedShorthand = { ...value };
```

Comment:
We can skip this copy now, since we're always spreading `...resolvedShorthand` below.
```suggestion
    resolvedShorthand = value;
```